dotCMS/src/main/java/com/dotcms/filters/NormalizationFilter.java

```diff
@@ -30,13 +37,35 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
 
             @Override
             public String getRequestURI() {
+                try {
+                    /* Normalization is the process of removing unnecessary "." and ".." segments from the path component of a hierarchical URI.
+                     1. Each "." segment is simply removed.
+                     2. A ".." segment is removed only if it is preceded by a non-".." segment.
+                     3. Normalization has no effect upon opaque URIs. (mailto:a@b.com)
+                     */
 
-                /* Normalization is the process of removing unnecessary "." and ".." segments from the path component of a hierarchical URI.
-                 1. Each "." segment is simply removed.
-                 2. A ".." segment is removed only if it is preceded by a non-".." segment.
-                 3. Normalization has no effect upon opaque URIs. (mailto:a@b.com)
-                 */
-                return URI.create(super.getRequestURI()).normalize().toString();
+                    String newNormal = URI.create(super.getRequestURI()).normalize().toString();
+                    newNormal = newNormal.startsWith(DOUBLEPEROIDS) ? newNormal.replace(DOUBLEPEROIDS, "") : newNormal;
+                    newNormal = newNormal.startsWith(SLASH) ? newNormal : SLASH + newNormal;
+
+                    // this should not happen 
+                    newNormal = newNormal.indexOf(QUESTION )>-1 ? newNormal.substring(0,newNormal.indexOf(StringPool.QUESTION)) : newNormal;
+                    while(newNormal.indexOf(DOUBLESLASH)>-1) {
+                        newNormal = newNormal.replace(DOUBLESLASH, SLASH);
+                    }
+                    return newNormal;
+                }
+                catch(IllegalArgumentException ill) {
+                    Logger.warnAndDebug(this.getClass(),ill);
+                    HttpServletResponse response= (HttpServletResponse) servletResponse;
+                    try {
+                        response.sendError(HttpServletResponse.SC_FORBIDDEN);
+                        response.flushBuffer();
+                    }
+                    catch(Exception e) {}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5436097)